projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php

```diff
@@ -0,0 +1,261 @@
+<?php
+/**
+ * Adds support for Jetpack Subscribe Modal feature
+ *
+ * @package automattic/jetpack-mu-wpcom
+ * @since $$next-version$$
+ */
+
+/**
+ * Jetpack_Subscribe_Modal class.
+ */
+class Jetpack_Subscribe_Modal {
+	/**
+	 * Jetpack_Subscribe_Modal singleton instance.
+	 *
+	 * @var Jetpack_Subscribe_Modal|null
+	 */
+	private static $instance;
+
+	/**
+	 * Jetpack_Subscribe_Modal instance init.
+	 */
+	public static function init() {
+		if ( self::$instance === null ) {
+			self::$instance = new Jetpack_Subscribe_Modal();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Jetpack_Subscribe_Modal class constructor.
+	 */
+	public function __construct() {
+		if ( $this->should_enable_subscriber_modal() ) {
+			add_action( 'wp_loaded', array( $this, 'wpcom_create_subscribe_template' ) );
+			add_action( 'wp_footer', array( $this, 'wpcom_add_subscribe_modal_to_frontend' ) );
+			add_action( 'wp_enqueue_scripts', array( $this, 'wpcom_enqueue_subscribe_modal_assets' ) );
+		}
+	}
+
+	/**
+	 * Adds a Subscribe Modal template part that can be edited in the editor.
+	 * This is later loaded in a pop up modal for Newsletter sites.
+	 *
+	 * @return void
+	 */
+	public function wpcom_create_subscribe_template() {
```

Comment:
I wonder if this could be triggered when the toggle is enabled in the Jetpack Settings. This may prove better for performance, since later on we wouldn't have to check for an existing post on each `wp_loaded`. What do you think?